log/gokit.go

```diff
@@ -6,101 +6,68 @@ package log
 
 import (
 	"fmt"
+	"io"
 	"os"
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 )
 
-// NewGoKitFormat creates a new Interface backed by a GoKit logger
-// format can be "json" or defaults to logfmt
-func NewGoKitFormat(l Level, f Format) Interface {
-	var logger log.Logger
-	if f.s == "json" {
-		logger = log.NewJSONLogger(log.NewSyncWriter(os.Stderr))
-	} else {
-		logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
-	}
-	return addStandardFields(logger, l)
-}
-
-// stand-alone for test purposes
-func addStandardFields(logger log.Logger, l Level) Interface {
-	logger = log.With(logger, "ts", log.DefaultTimestampUTC, "caller", log.Caller(5))
-	logger = level.NewFilter(logger, l.Gokit)
-	return gokit{logger}
-}
-
-// NewGoKit creates a new Interface backed by a GoKit logger
-func NewGoKit(l Level) Interface {
-	return NewGoKitFormat(l, Format{s: "logfmt"})
-}
-
-// GoKit wraps an existing gokit Logger.
-func GoKit(logger log.Logger) Interface {
-	return gokit{logger}
-}
-
-type gokit struct {
-	log.Logger
-}
-
-// Helper to defer sprintf until it is needed.
-type sprintf struct {
-	format string
-	args   []interface{}
-}
-
-func (s *sprintf) String() string {
-	return fmt.Sprintf(s.format, s.args...)
-}
+const (
+	LogfmtFormat = "logfmt"
+)
 
-// Helper to defer sprint until it is needed.
-// Note we don't use Sprintln because the output is passed to go-kit as one value among many on a line
-type sprint struct {
-	args []interface{}
+// NewGoKit creates a new GoKit logger with the given level, format and writer.
+// If the given writer is nil, os.Stderr is used.
+// If the given format is nil, logfmt is used.
+func NewGoKit(lvl Level, format string, writer io.Writer) log.Logger {
```

Comment:
Who needs this? I'm wondering why we need a logger without the "standard fields".